assets/javascripts/discourse/connectors/discovery-list-container-top/banner-ads.gjs

```diff
@@ -0,0 +1,32 @@
+import Component from "@glimmer/component";
+import { inject as service } from "@ember/service";
+import BannerAd from "../../components/banner-ad";
+
+const userHasBannerGroupsMembership = (bannerGroupIds, currentUser) => {
```

Comment:
I already commented on the serializer addition below but I think the group membership check will have to happen on the server

Reply:
Address in [DEV: Ensure we update cached categories](https://github.com/discourse/discourse-group-category-banner-ads/pull/1/commits/58fe93779d4315309af1423edea8747318cdde29)